redpanda/resources/resourcegroup/resource_resourcegroup.go

```diff
@@ -0,0 +1,182 @@
+// following the Terraform framework interfaces.
+package resourcegroup
+
+import (
+	"context"
+	"fmt"
+
+	controlplanev1beta2 "buf.build/gen/go/redpandadata/cloud/protocolbuffers/go/redpanda/api/controlplane/v1beta2"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/redpanda-data/terraform-provider-redpanda/redpanda/cloud"
+	"github.com/redpanda-data/terraform-provider-redpanda/redpanda/config"
+	"github.com/redpanda-data/terraform-provider-redpanda/redpanda/models"
+	"github.com/redpanda-data/terraform-provider-redpanda/redpanda/utils"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var (
+	_ resource.Resource                = &ResourceGroup{}
+	_ resource.ResourceWithConfigure   = &ResourceGroup{}
+	_ resource.ResourceWithImportState = &ResourceGroup{}
+)
+
+// ResourceGroup represents a cluster managed resource.
+type ResourceGroup struct {
+	CpCl *cloud.ControlPlaneClientSet
+}
+
+// Metadata returns the full name of the ResourceGroup resource.
+func (*ResourceGroup) Metadata(_ context.Context, _ resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = "redpanda_resourcegroup"
+}
+
+// Configure uses provider level data to configure ResourceGroup client.
+func (n *ResourceGroup) Configure(_ context.Context, request resource.ConfigureRequest, response *resource.ConfigureResponse) {
+	if request.ProviderData == nil {
+		// We can't add a diagnostic for an unset ProviderData here because
+		// during the early part of the terraform lifecycle, the provider data
+		// is not set and this is valid, but we also can't do anything until it
+		// is set.
+		return
+	}
+
+	p, ok := request.ProviderData.(config.Resource)
+	if !ok {
+		response.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *provider.Data, got: %T. Please report this issue to the provider developers.", request.ProviderData),
+		)
+		return
+	}
+	n.CpCl = cloud.NewControlPlaneClientSet(p.ControlPlaneConnection)
+}
+
+// Schema returns the schema for the ResourceGroup resource.
+func (*ResourceGroup) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = resourceGroupSchema()
+}
+
+// resourceGroupSchema defines the schema for a resource group.
+func resourceGroupSchema() schema.Schema {
+	return schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"name": schema.StringAttribute{
+				Required:      true,
+				Description:   "Name of the resource group. Changing the name of a resource group will result in a new resource group being created and the old one being destroyed",
```

Comment:
What happens to the resources within it? do they get destroyed too? are they reassigned to another resource group?

Reply:
The API manages this, it will fail with a "precondition not met".